### PR TITLE
Lesson 1.4

### PR DESCRIPTION
**Pull request by secch97**
Pull Request #4

- Refactored Todo List into a state.
- `<AddTodoForm/>` input text controlled.
- Add `'new todo'` to List.
- Refactored and cleaned up functional components. Applied more object destructuring and spread & rest operators.
- Added `prop-types` to current functional components.
- Started ToDoApp CSS Styling: began styling navigation bar.
- Added font awesome library.
- Fixed CSS Styling behaviour. 
- Added fontawesome react components to the project.
- Created a new functional component: `<NavigationBar>` which holds the component `<AddTodoForm>`.
- Removed inner text from label to use an icon instead.
- Added CSS Styling to the To Do List.
- Created <Footer> functional component. Styled footer.
- Added `scrollToTop()` function for a button.
- Added animations.
- Added media queries.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,33 +1,46 @@
 import React, { useState } from 'react';
+// Components
 import TodoList from './components/TodoList';
-import AddTodoForm from './components/AddTodoForm';
+import NavigationBar from './components/NavigationBar';
+import Footer from './components/Footer';
+// Helpers
+import { scrollToTop } from './helpers/scrollToTop';
```
Comment (viniciuspietscher) [inline src/App.jsx:7]: :1st_place_medal:

File: src/App.jsx [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
 
   /*
     ============================
     =         HANDLERS         =
     ============================
   */
-  const handleAddTodo = (todoTitle) =>{
-    setNewTodo(todoTitle);
+  const handleAddTodo = (newTodo) => {
+    setTodoList((todoList) => [...todoList, newTodo]);
```
Comment (viniciuspietscher) [inline src/App.jsx:25]: :100:

File: src/App.jsx [hunk 3]
```diff
@@ -1,33 +1,46 @@
...
+        <header>
+          <NavigationBar onAddTodo={handleAddTodo}/>
+        </header>
+        <main>
+          <TodoList todoList={todoList}/>
+        </main>
+        <Footer/>
+        <a className='button-top' onClick={scrollToTop}>
+          <FontAwesomeIcon icon={["fas", "circle-up"]} size="2x"/>
+        </a>
```
Comment (viniciuspietscher) [inline src/App.jsx:40]: That's pretty cool! Looks awesome too! :1st_place_medal:

File: src/components/AddTodoForm.jsx [hunk 4]
```diff
@@ -1,37 +1,74 @@
...
 };
+
+/*
+  ============================
+  =        PROP TYPES        =
+  ============================
+*/
+AddTodoForm.propTypes = {
+  onAddTodo: PropTypes.func.isRequired
+};
```
Comment (viniciuspietscher) [inline src/components/AddTodoForm.jsx:74]: I thought the propTypes were way ahead in the book haha! :grin:

File: src/components/TodoListItem.jsx [hunk 5]
```diff
@@ -1,11 +1,22 @@
 import React from 'react';
+// Third party libraries:
+import PropTypes from 'prop-types';
 
-const TodoListItem = ({toDo}) => {
+const TodoListItem = ({title}) => {
   return (
-    <li>{toDo.title}</li>
+    <li className='animate__animated animate__backInUp animate__fast'>{title}</li>
```
Comment (viniciuspietscher) [inline src/components/TodoListItem.jsx:7]: Looks awesome! :100:

File: src/helpers/scrollToTop.js [hunk 6]
```diff
@@ -0,0 +1,9 @@
+const scrollToTop = () => {
+    window.scrollTo({
+        top: 0
+    });
+};
```
Comment (viniciuspietscher) [inline src/helpers/scrollToTop.js:5]: :100:

**Review by viniciuspietscher — APPROVED** [review]
Great job! :1st_place_medal:

File: src/App.jsx (hunk 3 above)
Comment (secch97) [reply src/App.jsx:40]: > That's pretty cool! Looks awesome too! 🥇

Thanks!!

File: src/components/AddTodoForm.jsx (hunk 4 above)
Comment (secch97) [reply src/components/AddTodoForm.jsx:74]: > I thought the propTypes were way ahead in the book haha! 😁

Really? I wanted to see if I could validate react component's prop and I eventually found out about propTypes which was pretty useful to what I wanted haha, I did not know it was covered in the book way ahead

File: src/components/TodoListItem.jsx (hunk 5 above)
Comment (secch97) [reply src/components/TodoListItem.jsx:7]: > Great job! 🥇

Thank you Vinicius!

File: src/helpers/scrollToTop.js (hunk 6 above)
Comment (secch97) [reply src/helpers/scrollToTop.js:5]: > 💯

Thank you Vinicius!

**Comment by secch97** [comment]
> Great job! 🥇

Thank you Vinicius!!

File: src/components/AddTodoForm.jsx (hunk 4 above)
Comment (viniciuspietscher) [reply src/components/AddTodoForm.jsx:74]: Have you take a look at Typescript yet?

Comment (secch97) [reply src/components/AddTodoForm.jsx:74]: I haven't yet! Based on the name I suppose it is a type of JS focused on variable types, isn't it? 🤔

**Pull request merged**